### PR TITLE
Heretics Resource: Favor

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Heretics have a 'resource', Favor. Meaning heretics no longer gain charges in the Codex Cicatrix from sacrifices/tears in reality, but instead gain it themselves and can use it with any book. Adjusted costs and prices of reality tears. Also, heretics can read the Codex Gigas to gain more Favor, for a cost.

## Why It's Good For The Game

Further improves the heretics' abilities to function without a book. 
Allows heretics to reading the codex gigas, a stealthy way to gain favor and a more interesting objective.

## Changelog
:cl:
add: Heretics gain 'favor', replacing codex cicatrix charges.
add: Heretics can now read Codex Gigas to earn favor.
/:cl:

### CODE REVIEW

File: code/modules/library/lib_codex_gigas.dm
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "A shrill scream pains your ears..."

File: code/modules/library/lib_codex_gigas.dm
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "You get a sharp headache..."

File: code/modules/library/lib_codex_gigas.dm
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "You feel a stabbing pain in your head!"

File: code/modules/library/lib_codex_gigas.dm
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "Your heart burns!"

File: code/modules/library/lib_codex_gigas.dm
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm:160]: Maybe something more descriptive like: "Your body starts to deform!"

**Review by VictorVow — DISMISSED** [review]

**Comment by VictorVow** [comment]
Cool concept and definitely hard to cheese.

**Comment by eeSPee** [comment]
Thanks for the feedback! Updated!

File: code/modules/library/lib_codex_gigas.dm
Comment (eeSPee) [reply code/modules/library/lib_codex_gigas.dm:160]: Still thinking about this one...

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm
Comment (utilbird) [inline code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm]: You multiplied every other cost by 4 but this was 3. Was that intentional?

File: code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm
Comment (utilbird) [inline code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm]: above comment applies here too

File: code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm
Comment (heepox) [inline code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm]: I do believe this is intentional that its 9 instead of 12, but i could be wrong

Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm]: No I think that's my mistake.

File: html/changelog.html
Comment (park66665) [inline html/changelog.html]: Changelog changes?

```suggestion
```

File: html/changelogs/AutoChangeLog-pr-3002.yml
Comment (park66665) [inline html/changelogs/AutoChangeLog-pr-3002.yml]: You should delete this file.

**Comment by eeSPee** [comment]
I honestly don't know why github likes to stealth-update changelog.html and why there is no easy way to revert it.

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm:73]: Do not remove `cost`.

File: code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm:48]: ditto.

**Review by park66665 — COMMENTED** [review]
Also, can you remove now-obsolete `var/charge = 1` at the definition of `/obj/item/forbidden_book`?

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm:73]: Thanks. Must have removed it by mistake when fixing conflicts.

File: code/modules/library/lib_codex_gigas.dm
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm]:
```suggestion
			if (1, 2)	//light
```

**Comment by bloons3** [comment]
> Further improves the heretics' abilities to function without a book.

_Why?_

**Comment by TeomanTheGreat** [comment]
> > Further improves the heretics' abilities to function without a book.
> 
> _Why?_

Validhunting sec that's why. If your book is lost your done.

File: code/modules/library/lib_codex_gigas.dm
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm:119]:
```suggestion
			if (0 to 2)	//light
```

File: code/modules/library/lib_codex_gigas.dm
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm:32]: Codex *Gigas*? I thought it was the eldritch book: Codex Cicatrix. Is there any reason why you chose this book?

**Review by park66665 — CHANGES_REQUESTED** [review]
Screenshot 2021-02-25 221259

> **Charges** left

File: code/modules/library/lib_codex_gigas.dm
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm:116]:
```suggestion
	if(do_after(heretic, 10 SECONDS, target = src))
```

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_book.dm:21]:
```suggestion
```

#3450

**Comment by bloons3** [comment]
> Further improves the heretics' abilities to function without a book.

Heretics already have a mechanic to create a replacement book.
Moving the heretic points into the mind instead of the book means that there is no punishment for losing the book.
`If you lose your codex, you can make a new one by transmuting a bible, a pair of eyes, and some human skin. (better hope the chef will let you use the gibber)`
Unsure of what this PR is trying to accomplish...

**Comment by park66665** [comment]
Well, changing charges to favors only prevents unspent points being lost when a heretic loses a book. (It certainly is a small buff)

Another concern that has been raised was that the read-to-gain-favor being cheese-able albeit with a lot of efforts. I suggest a higher bracket for `dread` with even more drastic and dangerous effects, e.g. dismembering, burning the heretic, explosion, burning the book down, brain trauma, losing favor instead... you get the idea.

**Comment by eeSPee** [comment]
> Well, changing charges to favors only prevents unspent points being lost when a heretic loses a book. (It certainly is a small buff)
> 
> Another concern that has been raised was that the read-to-gain-favor being cheese-able albeit with a lot of efforts. I suggest a higher bracket for `dread` with even more drastic and dangerous effects, e.g. dismembering, burning the heretic, explosion, burning the book down, brain trauma, losing favor instead... you get the idea.

I mean, the main reason for this 'favor' to exist is so that Codex reading can exist.

Codex reading is a stealthier, less obvious alternative to reality hunting.

**Comment by park66665** [comment]
> I mean, the main reason for this 'favor' to exist is so that Codex reading can exist.

> Codex reading is a stealthier, less obvious alternative to reality hunting.

Your response is wrong at two levels:

1. You didn't reply to my second paragraph.
2. You didn't reply to my first paragraph properly either; you can make book reading work via moving book-reading feature from Codex Gigas to Codex Cicatrix, so moving charges to favor is not related to book reading. You also can search for book that the reader is holding.

You also didn't reply to my comment about Codex Gigas being used, which is counter-intuitive at best.

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm
Comment (park66665) [reply code/modules/antagonists/eldritch_cult/eldritch_book.dm:21]: UPDATE: This will be addressed via a separate PR: #3757

**Comment by eeSPee** [comment]
> 1. You didn't reply to my second paragraph.

That's a good idea. I didn't want it to be capital crime. It's supposed to be taxing, but not seriously crippling them. Because what you suggested will leave them crippled and exposed if they make that mistake.

> 2. You didn't reply to my first paragraph properly either; you can make book reading work via moving book-reading feature from Codex Gigas to Codex Cicatrix, so moving charges to favor is not related to book reading. You also can search for book that the reader is holding.

That could work, if the heretic datum keeps tabs on the dread.
 
> You also didn't reply to my comment about Codex Gigas being used, which is counter-intuitive at best.

I thought of this as a extra way for heretics to gain resource without needing the tears in reality, or to catch up if the tears of reality are used up.

If the Codex Cicatrix does this, then every heretic will be able to 'read' risk free from the book which could work, but was not what I had in mind.

**Comment by park66665** [comment]
> That's a good idea. I didn't want it to be capital crime. It's supposed to be taxing, but not seriously crippling them. Because what you suggested will leave them crippled and exposed if they make that mistake.

Crippling the players is the point. The reason why I am asking for this is the possibility of players that would cheese the side-effects through various strategies and reading Codex Gigas til Ascension. This is going to be hilarious and not very desirable. You should ensure that Codex reading stays a non-dominant way to earn Favor: no heretics moving out to space with a Codex Gigas and returning as an ascended heretic.

I won't merge this PR at this state if you don't convince me enough of the non-necessity of higher `dread` bracket.

> If the Codex Cicatrix does this, then every heretic will be able to 'read' risk free from the book which could work, but was not what I had in mind.

It is nothing but an *easy* winner-takes-all race where librarians get a crushing home advantage that has been improvised to take read-for-favor scheme out of Codex Cicatrix. I honestly am neutral on shifting this on something other than Codex Cicatrix, but if you don't understand the implications of it, I won't look at this PR favorably. This is neither stealthy nor interesting; you should know that.

**Comment by TeomanTheGreat** [comment]
Just make it so curator can't be a heretic

**Comment by TeomanTheGreat** [comment]
> > That's a good idea. I didn't want it to be capital crime. It's supposed to be taxing, but not seriously crippling them. Because what you suggested will leave them crippled and exposed if they make that mistake.
> 
> Crippling the players is the point. The reason why I am asking for this is the possibility of players that would cheese the side-effects through various strategies and reading Codex Gigas til Ascension. This is going to be hilarious and not very desirable. You should ensure that Codex reading stays a non-dominant way to earn Favor: no heretics moving out to space with a Codex Gigas and returning as an ascended heretic.
> 
> I won't merge this PR at this state if you don't convince me enough of the non-necessity of higher `dread` bracket.
> 
> > If the Codex Cicatrix does this, then every heretic will be able to 'read' risk free from the book which could work, but was not what I had in mind.
> 
> It is nothing but an *easy* winner-takes-all race where librarians get a crushing home advantage that has been improvised to take read-for-favor scheme out of Codex Cicatrix. I honestly am neutral on shifting this on something other than Codex Cicatrix, but if you don't understand the implications of it, I won't look at this PR favorably. This is neither stealthy nor interesting; you should know that.

Just make it so curator can't be a heretic. It gives him the job of actually protecting the codex instead of just doing random shit.

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm
Comment (Crossedfall) [reply code/modules/antagonists/eldritch_cult/eldritch_book.dm:21]:
```suggestion
	. += "The Tome holds [charge] charges."
```

**Comment by ike709** [comment]
Stale

**Pull request closed without merging**
